packages/manifold/contracts/edition/ManifoldERC721Edition.sol

```diff
@@ -134,22 +128,23 @@ contract ManifoldERC721Edition is CreatorExtension, ICreatorExtensionTokenURI, I
             indexRanges.push(IndexRange(startIndex, count));
           }
         }
-        _totalSupply[creator][series] += count;
+        _totalSupply[instanceId] += count;
     }
 
     /**
      * @dev Index from tokenId
      */
-    function _tokenSeriesAndIndex(address creator, uint256 tokenId) internal view returns(uint256, uint256) {
-        require(_currentSeries[creator] > 0, "Invalid token");
-        for (uint series=1; series <= _currentSeries[creator]; series++) {
-            IndexRange[] memory indexRanges = _indexRanges[creator][series];
+    function _tokenInstanceAndIndex(address creator, uint256 tokenId) internal view returns(uint256, uint256) {
+        // Go through all their series until we find the tokenId
+        for (uint i = 0; i < _creatorInstanceIds[creator].length; i++) {
```

Comment:
uint256 i; i < < _creatorInstanceIds[creator].length;

always use uint256 (be explicit)

Move ++i to unchecked { ++i; } at bottom of loop to save gas.